File: server/src/main/java/com/mio/server/models/Board.java
package com.mio.server.models;

import com.mio.server.builders.BoardBuilder;

public class Board {

    private final Point point;
    private final String type;

    public Board(BoardBuilder builder) {
        this.point = builder.getPoint();
        this.type = builder.getType();
    }

    public Point getPoint() {
        return point;
    }

    public String getType() {
        return type;
    }

    @Override
    public String toString() {
        return "Board{" +
                "point=" + point +
                ", type=" + type +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Board board = (Board) o;

        if (!point.equals(board.point)) return false;
        return type.equals(board.type);
    }

    @Override
    public int hashCode() {
        int result = point.hashCode();
        result = 31 * result + type.hashCode();
        return result;
    }
}

File: Boxworld/app/src/main/java/com/mio/cliente/boxworld/models/Board.kt
package com.mio.cliente.boxworld.models

import com.mio.cliente.boxworld.builders.BoardBuilder
import java.io.Serializable

data class Board(private val builder: BoardBuilder): Serializable {

    val point = builder.point
    val type: String = builder.type
    var hasBox: Boolean = false
    var hasTarget: Boolean = false
    var hasPlayer: Boolean = false


}